File: src/main/java/mil/nga/tiff/field/type/ByteField.java
package mil.nga.tiff.field.type;

import java.nio.ByteBuffer;

abstract sealed class ByteField<T extends Number> extends NumericFieldType<T> permits UnsignedByteField, SignedByteField {

    @Override
    final protected void writeSample(ByteBuffer buffer, T value) {
        buffer.put(value.byteValue());
    }

    @Override
    final public void transferSample(ByteBuffer outBuffer, ByteBuffer inBuffer) {
        outBuffer.put(inBuffer.get());
    }

}

File: src/main/java/mil/nga/tiff/field/type/ByteField.kt
package mil.nga.tiff.field.type

import java.nio.ByteBuffer

abstract class ByteField<T : Number> : NumericFieldType<T>() {
    override fun writeSample(buffer: ByteBuffer, value: T) {
        buffer.put(value.toByte())
    }

    override fun transferSample(outBuffer: ByteBuffer, inBuffer: ByteBuffer) {
        outBuffer.put(inBuffer.get())
    }
}
